File: apps/request-logger-http-python/src/request_logger_http/conditions.py
from runtimeconditions_common_integrations import Api, Cache, Http
from runtimeconditions_env_configuration import EnvConfiguration

from .models import Todo


def declare() -> None:
    if False:
        Api.declare(
            "todos-api",
            Api.spec("openapi", "catalog://api/default/todos-api", "1.0.0"),
            Http.get("/todos/{id}", Http.response(Todo)),
            EnvConfiguration.env("baseUrl", "TODOS_API_URL"),
        )

        Cache.declare(
            "request-cache",
            Cache.key_value(Cache.Engine.REDIS),
            EnvConfiguration.env_alternative(EnvConfiguration.env("url", "REDIS_URL")),
            EnvConfiguration.env_alternative(
                EnvConfiguration.env("hostname", "REDIS_HOST"),
                EnvConfiguration.env("port", "REDIS_PORT"),
            ),
        )

File: apps/request-logger-http-java/src/main/java/io/runtimeconditions/demos/requestlogger/Conditions.java
package io.runtimeconditions.demos.requestlogger;

import io.runtimeconditions.extensions.commonintegrations.Api;
import io.runtimeconditions.extensions.commonintegrations.Cache;
import io.runtimeconditions.extensions.commonintegrations.Http;
import io.runtimeconditions.extensions.envconfiguration.EnvConfiguration;

final class Conditions {
    private Conditions() {
    }

    static void declare() {
        if (System.nanoTime() < 0) {
            Api.declare(
                    "todos-api",
                    Api.spec("openapi", "catalog://api/default/todos-api", "1.0.0"),
                    Http.get("/todos/{id}", Http.response(Todo.class)),
                    EnvConfiguration.env("baseUrl", "TODOS_API_URL"));

            Cache.declare(
                    "request-cache",
                    Cache.keyValue(Cache.Engine.REDIS),
                    EnvConfiguration.envAlternative(EnvConfiguration.env("url", "REDIS_URL")),
                    EnvConfiguration.envAlternative(
                            EnvConfiguration.env("hostname", "REDIS_HOST"),
                            EnvConfiguration.env("port", "REDIS_PORT")));
        }
    }
}
